db/migrate/20191126181009_seed_test_results.rb

```diff
@@ -13,24 +15,23 @@ def up
       rule_results_by_end_time = scan_results(host, profile)
         .group_by(&:end_time)
       rule_results_by_end_time.each do |end_time, rule_results|
-        test_result = TestResult.create!(
-          host: host,
-          profile: profile
+        rules_passed = rule_results.count { |rr| RuleResult::PASSED.include?(rr.result) }
+        rules_failed = rule_results.count { |rr| RuleResult::FAIL.include?(rr.result) }
+        rule_results_set << rule_results
+        test_results << TestResult.new(
+          host_id: host.id,
+          profile_id: profile.id,
+          start_time: rule_results[0].start_time || Time.now.utc,
+          end_time: rule_results[0].end_time || Time.now.utc,
+          score: compliance_score(rules_passed, rules_failed)
         )
-        rule_result_ids = rule_results.pluck(:id)
-        puts("   - Number of RuleResults found: #{rule_result_ids.count}"\
-             " at time #{end_time}")
-        latest_rule_results = RuleResult.where(id: rule_result_ids)
-        latest_rule_results.update_all(test_result_id: test_result.id)
-        test_result.update!(
-          start_time: latest_rule_results[0].start_time || Time.now.utc,
-          end_time: latest_rule_results[0].end_time || Time.now.utc,
-          score: compliance_score(host.rules_passed(profile),
-                                  host.rules_failed(profile))
-        ) unless latest_rule_results.empty?
       end
       migrated += 1
     end
+    imported_test_results = TestResult.import test_results
+    imported_test_results.ids.zip(rule_results_set).each do |imported_test_result, rule_results|
+      RuleResult.where(id: rule_results.pluck(:id)).update_all(test_result_id: imported_test_result)
```

Comment:
This line seems to be a bottleneck during my testing. We could probably use more SQL here rather than pluck if we want to further optimize. There's also some danger here of hitting the max SQL query length.